### PR TITLE
Unique cleanup

### PR DESCRIPTION
**Pull request by tmccombs**
Clean up a couple of deprecated features used in Unique.

### CODE REVIEW

File: std/typecons.d [hunk 1]
```diff
@@ -146,7 +146,12 @@ public:
     ~this()
     {
         debug(Unique) writeln("Unique destructor of ", (_p is null)? null: _p);
-        if (_p !is null) delete _p;
+        if (_p !is null)
+        {
+            import core.memory : GC;
+            destroy(_p);
+            GC.free(cast(void*) _p);
```
Comment (tmccombs) [inline std/typecons.d]: I'm not entirely sure if the GC.free should be here. It seems consistent with the previous behavior, and my expectation would be that Unique should free the memory it is holding when it goes out of scope, but opAssign only calls `destroy` not `GC.free`.

Comment (9il) [inline std/typecons.d]: `destroy` is enough here

File: std/typecons.d [hunk 2]
```diff
@@ -163,8 +163,9 @@ public:
         debug(Unique) writeln("return from Release");
         return u;
     }
+
     /** Forwards member access to contents. */
-    auto opDot() inout { return _p; }
+    mixin Proxy!_p;
```
Comment (JackStouffer) [inline std/typecons.d]: Pretty sure Proxy can only be used with lvalues, but p can be an lvalue or an rvalue

File: std/typecons.d [hunk 3]
```diff
@@ -1744,7 +1745,7 @@ Rebindable!T rebindable(T)(Rebindable!T obj)
     obj2 = new immutable(C);
     assert(obj1.get !is null);
 
-    // test opDot
+    // test dispatch
```
Comment (JackStouffer) [inline std/typecons.d]: Why are you modifying the tests for Rebindable?

**Comment by andralex** [comment]
@tmccombs this is getting there, could you respond to @JackStouffer and take this to the finish line? Thx!

File: std/typecons.d (hunk 2 above)
Comment (tmccombs) [inline std/typecons.d]: `_p` is a member variable. In what circumstances would it be an rvalue?

**Comment by andralex** [comment]
Auto-merge toggled on

**Pull request merged**